DNN Platform/Library/Entities/Tabs/TabVersions/TabVersionController.cs

```diff
@@ -112,6 +117,26 @@ private static string GetTabVersionsCacheKey(int tabId)
         {
             return string.Format(DataCache.TabVersionsCacheKey, tabId);
         }
+        
+        protected virtual TimeZoneInfo GetDatabaseDateTimeOffset()
+        {
+            var dateTimeOffset = DateUtils.GetDatabaseDateTimeOffset();
+            var offset = dateTimeOffset.Offset;
+            var id = string.Format("UTC {0}", offset.ToString());
+            return TimeZoneInfo.CreateCustomTimeZone(id, offset, id, id);
+        }
+
+        private IEnumerable<TabVersion> ApplyUserLocalTime(IEnumerable<TabVersion> tabVersions, TimeZoneInfo userTimeZone)
+        {
+            var serverTimeZone = GetDatabaseDateTimeOffset();
```

Comment:
why not just using TimeZoneInfo.Convert to convert from one timezone to another?

Reply:
I second this thought

Reply:
What if Database is on a remote server with a different TZ?

Reply:
@sleupold @mitchelsellers @tpluscode 
We also have to consider the CreatedOnDate property, inherited from BaseEntityInfo, is a read only property. It's never set from code. Its value comes from a database "NOW()" function. That's reason why I took care of using Database timezone offset instead of the server's one. Whatever we do with that *CreatedOnDate* property, it's meaningful only if we know the Database locale time settings. IMHO.

Reply:
@mean2me 
Timestamps should always be obtained from SQL Server, therefore there is a function to get it in server time or UTC. In general, UTC is the better choice, however, for historical reasons, DNN always uses local SQL Server time.